### ISSUE TITLE
(Some) expressions with symbolic order derivative cannot be simplified without an error

### ISSUE THREAD

**Issue by oscargus** [sympy/sympy#22210]
From #22200 

Edited to minimum non-working example
```
from sympy import *
x = Symbol('x')
d = Symbol('d', integer=True)
expr = 2*Derivative(sin(x), (x, d))  # Multiplication required
expr.simplify()
```
leads to
```
Traceback (most recent call last):

  File "C:\Users\Oscar\AppData\Local\Temp/ipykernel_28316/3646803567.py", line 5, in <module>
    expr.simplify()

  File "C:\Users\Oscar\sympy\sympy\core\basic.py", line 1686, in simplify
    return simplify(self, **kwargs)

  File "C:\Users\Oscar\sympy\sympy\simplify\simplify.py", line 635, in simplify
    expr = Mul(*powsimp(expr).as_content_primitive())

  File "C:\Users\Oscar\sympy\sympy\simplify\powsimp.py", line 160, in powsimp
    for b, e in ordered(iter(c_powers.items())):

  File "C:\Users\Oscar\sympy\sympy\core\compatibility.py", line 614, in ordered
    yield from d[k]

  File "C:\Users\Oscar\sympy\sympy\core\compatibility.py", line 595, in ordered
    d[f(a)].append(a)

  File "C:\Users\Oscar\sympy\sympy\core\compatibility.py", line 501, in _nodes
    return 1 + sum(_nodes(ei) for ei in e)

  File "C:\Users\Oscar\sympy\sympy\core\compatibility.py", line 501, in <genexpr>
    return 1 + sum(_nodes(ei) for ei in e)

  File "C:\Users\Oscar\sympy\sympy\core\compatibility.py", line 498, in _nodes
    return _nodes(e.expr) + len(e.variables)

  File "C:\Users\Oscar\sympy\sympy\core\function.py", line 1662, in variables
    raise TypeError(filldedent('''

TypeError: 
Cannot give expansion for symbolic count. If you just want a list of
all variables of differentiation, use _wrt_variables.
```

**Comment by oscargus** [comment]
There are some comments illustrating that this is a known issue (to some extent):
https://github.com/sympy/sympy/blob/570ad001d7f5a3c733c54635fec90edb8f778443/sympy/core/function.py#L1656-L1658

**Comment by smichr** [comment]
That's why it should be deprecated: it doesn't give a universally applicable result. Until then,
```diff
diff --git a/sympy/core/compatibility.py b/sympy/core/compatibility.py
index 8ddd83e750..f41f239504 100644
--- a/sympy/core/compatibility.py
+++ b/sympy/core/compatibility.py
@@ -495,7 +495,8 @@ def _nodes(e):

     if isinstance(e, Basic):
         if isinstance(e, Derivative):
-            return _nodes(e.expr) + len(e.variables)
+            return _nodes(e.expr) + sum(i[1] if i[1].is_Number else
+                _nodes(i[1]) for i in e.variable_count)
         return _node_count(e)
     elif iterable(e):
         return 1 + sum(_nodes(ei) for ei in e)
```
gives
```python
>>> from sympy.core.compatibility import _nodes as f
>>> [f(Derivative(sin(x),(x,i))) for i in (y, 1, 2, y + 1)]
[3, 3, 4, 5]
>>> d = Symbol('d', integer=True)
>>> expr = 2*Derivative(sin(x), (x, d))  # Multiplication required
>>> expr.simplify()
2*Derivative(sin(x), (x, d))
```

**Comment by smichr** [comment]
fix in #22224
